docs/tools/config_system.md

```diff
@@ -1,23 +1,135 @@
 ## The configuration system
 
-The Arm Mbed configuration system customizes the compile time configuration of various Mbed components (targets, libraries and applications). Each component can define a number of *configuration parameters*. The values of these configuration parameters can then be *overridden* in various ways. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
+The Arm Mbed OS configuration system, a part of the Arm Mbed OS Build Tools, customizes compile time configuration parameters. Each library may define a number of configuration parameters in its `mbed_lib.json`. The values of these configuration parameters may be overridden by `mbed_app.json`. Configuration is defined using [JSON](http://www.json.org/). Some examples of configuration parameters:
```

Comment:
Consider clarifying what you mean by "library" here, since you're talking about neither static nor dynamic libraries, nor `.lib` files understood by Mbed CLI. Maybe even just do it in a footnote.

Reply:
I'll add a Note: library in this context is anything with an `mbed_lib.json`.

It's a little circular XD